scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SynchronousSchedulerAgent.java

```diff
@@ -169,7 +173,10 @@ public void schedule(final ScheduleRequest request, final ResultsProtocol.Writer
         }
       }
       if (!failedGoals.isEmpty()) {
-        writer.failWith(failedGoals.toString());
+        writer.failWith(b -> b
+            .type("SCHEDULING_GOALS_FAILED")
```

Comment:
It's not ideal that these `type` fields are hardcoded string literals but I figured this is pretty innocuous right now as every `failWith()` invocation is within this method – with the exception of one `failWith()` within the `ThreadedSchedulerAgent` that is called as a catch-all for any `Throwable`.

Reply:
> there's only ever one failWith() call per failure type
I'm in favor of this property because I think it will help make the errors more traceable 👍